app/src/main/java/com/selfdrivingboat/boatcontroller/MainActivity.java

```diff
@@ -265,19 +309,132 @@ protected void onPostExecute(Boolean result) {
                 temperature = 0;
                 temprature_count = 0;
             }
+        } else if (event.sensor.getType() == Sensor.TYPE_MAGNETIC_FIELD) {
+            this.logger.i("[androidMagneticfield] onSensorChanged");
+            magnetic_x = event.values[0];
+            magnetic_y = event.values[1];
+            magnetic_z = event.values[2];
+            magnetometercount_count += 1;
+            if (magnetometercount_count == 2) {
+                class sendDataTask extends AsyncTask<Void, Void, Boolean> {
+                    @Override
+                    protected Boolean doInBackground(Void... voids) {
+                        InfluxDBWrites.sendMPU6050magnetic_field(magnetic_x, magnetic_y, magnetic_z);
+                        return true;
+                    }
+
+                    protected void onPostExecute(Boolean result) {
+                        if (result) {
+                            MainActivity.this.logger.i(" [androidMagneticfield] data sent to influx");
+                        } else {
+                            MainActivity.this.logger.i("[androidMagneticfield] data sent to influxdb fail");
+                        }
+                    }
+                }
+                new sendDataTask().execute();
+                magnetic_x = 0;
+                magnetic_y = 0;
+                magnetic_z = 0;
+                magnetometercount_count = 0;
+            }
+        } else if (event.sensor.getType() == Sensor.TYPE_PROXIMITY) {

```

Comment:
wow! you wrote so much code. Can't believe 2 weeks ago you didn't know Java.
This looks good to me for now, something to keep in mind for the future. You never want to duplicate your code, i.e. avoid to copy paste as much as possible. If you are copy pasting something usually means that you can put stuff in function and use that function in more places. That is really good because when you will need to change the code you don't need to change it in all the places where you copy pasted but just in your function :) amazing job